pylint/checkers/variables.py

```diff
@@ -1543,7 +1543,10 @@ def _check_consumer(
                     )
 
         elif self._is_only_type_assignment(node, defstmt):
-            self.add_message("undefined-variable", args=node.name, node=node)
+            if node.scope().locals.get(node.name):
```

Comment:
I did not look in detail so it's an open question but performance wise would it make sense to create another function than `_is_only_type_assignment`  according to you? It seems it handle the two use case then we have to do another check to separate the use cases.

Reply:
I don't think it would be worth the effort, the two checkers would look pretty much identical, since they both would have to keep looking in outer scopes. I don't think it's much of a performance penalty to do one more hit to `scope()` since it's only when emitting a message.